torch/csrc/jit/mobile/backport_manager.cpp

```diff
@@ -146,14 +185,11 @@ void writeArchiveV4(
 }
 
 bool backport_v5_to_v4(
-    PyTorchStreamReader& reader,
-    PyTorchStreamWriter& writer) {
+    std::stringstream& input_model_stream,
```

Comment:
Since the output stream is created and swapped, why not we pass one stringstream (called io_model_stream), and make it as both input and output?

Reply:
Also I didn't see mobile_debug_handles.pkl. The version number is also written there, which should also be backported?

Reply:
> Since the output stream is created and swapped, why not we pass one stringstream (called io_model_stream), and make it as both input and output?

`stream` is tricky and it's really hard to debug if one byte is wrong. If using one parameter as input, it sort of indicates passing as reference and modify in place. I think it's better to make sure the output stream is new and clean. I thought about the other suggestion from you `std::stringstream backportFunction(const stringstream& input)`. I feel like it's still good to have a boolean return to indicate if the the backport is success. What do you think?

> Also I didn't see mobile_debug_handles.pkl. The version number is also written there, which should also be backported?

I didn't realize there is a `bytecode_version` number in `mobile_debug_handles.pkl`. Ideally it's better to have one `bytecode_version` number in a model in one place (`bytecode.pkl`). Let's see if we can remove the `bytecode_version` in `mobile_debug_handles.pkl`. cc: @kimishpatel

Reply:
For function interface, consider `stringstream backportFunction(const stringstream& input)`. Move the output stream as return value? 

The version number is emplaced in https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/serialization/export_module.cpp#L634 . Since it's a separate topic, if there's any update related to this file, it could be put in another PR.

Reply:
Yeah I talked to @kimishpatel about debug pkl
We need to map all the dependencies we have between serialized debug info, runtime, and symbolication and think about what versioning means:

* Changes in the generation of debug handles seem would be very disruptive, and not recoverable (perhaps?).
* Changes to serialized data: do they need a version bump?
* What changes do we need to version, and can we reuse the bytecode version for them?
... 

Anyway, something to address separately.

Reply:
I will put together some thoughts on versioning for debug info. In regards to debug handles pickling, some initial thoughts: I think it existed as separate archive as this was an optional feature we introduces and we did not want to pollute bytecode.pkl. It seemed cleaner that way. However if we are trying debug_handles and bytecode under same version then we should figure out putting them together. But that would be another BC breaking change.